vumi/message.py

```diff
@@ -27,12 +27,17 @@ def format_vumi_date(timestamp):
 def parse_vumi_date(value):
     """Parse a timestamp string using the Vumi date format.
 
+    Timestamps without microseconds are also parsed correctly.
+
     :param str value:
         The string to parse.
     :return datetime:
         A datetime object representing the timestamp.
     """
-    return datetime.strptime(value, VUMI_DATE_FORMAT)
+    date_format = VUMI_DATE_FORMAT
+    if "." not in value:
+        date_format = date_format.split(".")[0]
```

Comment:
`"." not in value` is O(`len(value)`), which isn't really all that expensive. (We've probably already spent at least that much time parsing JSON containing that value, so if there's a DoS attack it's already succeeded by the time we get here.) If we want to do micro-optimisations like this, we should take measurements and check that there's some benefit to counter the loss in clarity.